File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] a) {
        Arrays.sort(a, (i1, i2) -> i1[0] - i2[0]);
        int index = 0;

        for (int i = 1; i < a.length; i++) {
            if (a[i][0] <= a[index][1]) a[index][1] = Math.max(a[i][1], a[index][1]);
            else {
                index++;
                a[index] = a[i];
            }
        }
        return Arrays.copyOf(a, index + 1);
    }
}

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        ans=[]
        n=len(intervals)
        for start,end in intervals:
            if not ans or start>ans[-1][1]:
                ans.append([start,end])
            else:
                ans[-1][1]=max(ans[-1][1],end)
        return ans